scripts/source-maintenance.sh

```diff
@@ -312,7 +312,7 @@ processDebugSections ()
 {
     destination="doc/debug-sections.txt"
 
-    sort -u < doc/debug-sections.tmp | sort -n > doc/debug-sections.tmp2
+    LC_ALL=C sort -u < doc/debug-sections.tmp > doc/debug-sections.tmp2
```

Comment:
I suspect we should export `LC_ALL` for the entire scripts/source-maintenance.sh run, but the exact effects of that are difficult (for me) to predict, and I am not sure that should be done in the PR that is focusing on one specific file generation. I will do it of others think it is a good idea.

FWIW, I did check other "sort" statements in scripts/source-maintenance.sh and they should be more-or-less OK as far as locale is concerned because they are either used just to detect duplicates (in ASCII lines) or are used to sort based on (ASCII) numeric keys.

Reply:
Please restore the -n parameter here.

Reply:
I agree it would make sense and just bite the bullet. It might add changes once but then stability should be improved

Reply:
> Please restore the -n parameter here.

Restoring `-n` would reintroduce a bug this PR is fixing. As PR description says, the `-n` parameter was wrong -- the fields we are comparing here are _not_ numeric (they start with the word "section"). `sort --debug -n` confirms that `-n` results in a failure to find the numeric key (which is then substituted with the whole line).

----

While restoring `-n` would be wrong, we could specify multiple sorting keys so that one of them becomes numeric, but it is difficult to do that in a portable way using "sort". We would probably need to switch to Perl for that. We still have a few unused two-digit section numbers to go, so I would prefer not to spend time on that improvement right now. I actually meant to post a comment about that but forgot.

Reply:
@yadij, you left your "restore `-n`" change request in a thread about making `LC_ALL=C` global (i.e. applying to anything that source-maintenance.sh script starts and not just this sort command where it is known to be needed). AFAICT, @kinkie thinks that we should make `LC_ALL=C` global (in this PR). When you update your review, please comment whether you think we should make `LC_ALL=C` global (in this PR).

Reply:
Ah, okay removing my request for restore.


IMO we should use LC_ALL=C always when sorting.

Reply:
> IMO we should use LC_ALL=C always when sorting.

If by "when sorting" you mean "while executing the sort command", then that does not work well because "sort" is often used in a _combination_ with other locale-sensitive commands (e.g., "uniq"). Some of those commands may be used indirectly or in separate contexts -- they may be difficult to find. All such command combinations should use the same locale to produce correct outcome. Guessing which commands/functions/contexts/etc. need `LC_ALL=C` and sprinkling the script with `LC_ALL=C` expressions in various places is not a good solution IMO and is definitely outside this PR scope.

If by "when sorting" you mean "globally in any script that sorts something somewhere", then that is what "make `LC_ALL=C` global" proposal suggests. That proposal does not have the above mentioned flaws. If you support that proposal, and want to see it implemented in _this_ PR, then please say so, and I will modify this PR accordingly.

Please clarify what you mean.